### PR TITLE
Add Trino 426 release notes

### LINKED ISSUE #18802: Release notes for 426

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 426 release. Fixes #18802 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 25 Aug 2023

* #18807 ✅ rn ✅ docs
* #18810 ✅ rn ✅ docs
* #18804 ✅ rn ✅ docs
* #18813 ✅ rn ✅ docs

## 27 Aug 2023

* #18783 ✅ rn ✅ docs

## 28 Aug 2023

* #18825 ✅ rn ✅ docs
* #18274 ✅ rn ❌ docs
* #18834 ✅ rn ✅ docs
* #18717 ✅ rn ✅ docs
* #18827 ✅ rn ✅ docs
* #18836 ✅ rn ✅ docs
* #18729 ✅ rn ✅ docs
* #18784 ✅ rn ✅ docs
* #18662 ✅ rn ✅ docs
* #18788 ✅ rn ✅ docs
* #18333 ✅ rn ✅ docs
* #18840 ✅ rn ✅ docs

## 29 Aug 2023

* #18839 ✅ rn ✅ docs
* #18706 ✅ rn ✅ docs
* #18815 ✅ rn ✅ docs
* #18833 ✅ rn ✅ docs
* #18837 ✅ rn ✅ docs
* #18852 ✅ rn ✅ docs
* #18708 ✅ rn ✅ docs
* #18792 ✅ rn ✅ docs
* #18828 ✅ rn ✅ docs
* #18730 ✅ rn ✅ docs
* #18846 ✅ rn ✅ docs

## 30 Aug 2023

* #18693 ✅ rn ✅ docs
* #18806 ✅ rn ✅ docs
* #18849 ✅ rn ✅ docs
* #18847 ✅ rn ✅ docs
* #18866 ✅ rn ✅ docs
* #18870 ✅ rn ✅ docs
* #16067 ✅ rn ✅ docs
* #18868 ✅ rn ✅ docs
* #18538 ✅ rn ✅ docs
* #18856 ✅ rn ✅ docs

## 31 Aug 2023

* #18857 ✅ rn ✅ docs
* #18808 ✅ rn ✅ docs
* #18876 ✅ rn ✅ docs
* #18883 ✅ rn ✅ docs
* #18197 ✅ rn ✅ docs
* #18878 ✅ rn ✅ docs
* #18879 ✅ rn ✅ docs
* #18635 ✅ rn ✅ docs
* #18655 ✅ rn ✅ docs
* #18887 ✅ rn ✅ docs
* #18874 ✅ rn ✅ docs

## 1 Sep 2023

* #18880 ✅ rn ✅ docs
* #18892 ✅ rn ✅ docs
* #18893 ✅ rn ✅ docs
* #18721 ✅ rn ✅ docs
* #18888 ✅ rn ✅ docs
* #18803 ✅ rn ✅ docs
* #18890 ✅ rn ✅ docs

## 2 Sep 2023

* #18894 ✅ rn ✅ docs
* #18886 ✅ rn ✅ docs

## 4 Sep 2023

* #18904 ✅ rn ✅ docs
* #18902 ✅ rn ✅ docs
* #18905 ✅ rn ✅ docs
* #13994 ✅ rn ✅ docs
* #18915 ✅ rn ✅ docs

## 5 Sep 2023

* #18917 ✅ rn ✅ docs
* #18896 ✅ rn ✅ docs
* #18903 ✅ rn ✅ docs
* #18908 ✅ rn ✅ docs
* #18899 ✅ rn ✅ docs
* #18911 ✅ rn ✅ docs
* #18914 ✅ rn ✅ docs
* #18929 ✅ rn ✅ docs
* #18927 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-426.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+## Security
+
+* Fix authentication failure with OAuth 2.0 when authentication tokens are
+  larger than 4 KB. ({issue}`18836`)
+
+## Delta Lake connector
+
+* Add support for the `TRUNCATE TABLE` statement. ({issue}`18786`)
+* Add support for `CASCADE` option in `DROP SCHEMA` statements. ({issue}`18333`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-426.md:16]:
```suggestion
* Add support for the `CASCADE` option in `DROP SCHEMA` statements. ({issue}`18333`)
```

File: docs/src/main/sphinx/release/release-426.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* Fix writing an incorrect transaction log for partitioned tables with an `id`
+  or `name` column mapping mode. ({issue}`18661`)
+
+## Hive connector
+
+* Add the `hive.metastore.thrift.batch-fetch.enabled` configuration property,
+  which can be set to `false` to disable batch metadata fetching from the Hive
+  metastore. ({issue}`18111`)
+* Fix `ANALYZE` failure when row count stats are missing. ({issue}`18798`)
+* Fix the `hive.target-max-file-size` configuration property being ignored
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-426.md:26]: Fix ... being ignored .. is kinda odd. Did we do that in the past as well..

Alternatively we could do 

Honor the `hive.target-max-file-size` configuration property when writing to sorted tables

but thats also odd .. Take the ... into account? 

Or 

Avoid ignoring the property...

not sure overall

File: docs/src/main/sphinx/release/release-426.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  when writing to sorted tables. ({issue}`18653`)
+* Fix error when reading large SequenceFile, RCFile, or Avro files. ({issue}`18837`)
+
+## Iceberg connector
+
+* Fix the `iceberg.target-max-file-size` configuration property being ignored
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-426.md:32]: see above

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-426.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-426.md:26]: I think I like this best. I try to keep the first word to "Add/Support," "Improve," or "Fix" as often as possible so it's extremely clear what kind of change is happening. "Honor" is vague, "avoid ignoring" is a double negative, and I think I'd stand by this wording.

Comment (electrum) [reply docs/src/main/sphinx/release/release-426.md:26]: We could also phrase it as

> Support the `xxx` configuration property when writing to sorted tables.

If it never worked, then this is a newly added feature. Although users expect it to work since it wasn't documented to not work, so it's also kind of a fix...

Comment (electrum) [reply docs/src/main/sphinx/release/release-426.md:26]: I think what's weird about this phrasing is treating "xxx config property being ignored" as the target of what is being fixed. This is one of those cases where "Fixed an issue where the xxx was ignored" might be an improvement. But we always avoid the "fixed an issue" phrasing, so... I'm fine with the original.

File: docs/src/main/sphinx/release/release-426.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Hive connector
+
+* Add the `hive.metastore.thrift.batch-fetch.enabled` configuration property,
+  which can be set to `false` to disable batch metadata fetching from the Hive
+  metastore. ({issue}`18111`)
+* Fix `ANALYZE` failure when row count stats are missing. ({issue}`18798`)
+* Fix the `hive.target-max-file-size` configuration property being ignored
+  when writing to sorted tables. ({issue}`18653`)
+* Fix error when reading large SequenceFile, RCFile, or Avro files. ({issue}`18837`)
```
Comment (electrum) [inline docs/src/main/sphinx/release/release-426.md:28]: The original "Fix reading ..." seems more clear. If we want to expand it, "Fix query failure when reading" is better.

**Review by electrum — APPROVED** [review]

File: docs/src/main/sphinx/release/release-426.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# Release 426 (5 Sep 2023)
+
+## General
+
+* Add support for `SET SESSION AUTHORIZATION` and `RESET SESSION AUTHORIZATION`
+  syntax. ({issue}`16067`)
+* Add support for automatic type coercion when creating tables. ({issue}`13994`)
+* Improve performance of decimal aggregations. ({issue}`18868`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-426.md:8]: "aggregations over decimal values"

File: docs/src/main/sphinx/release/release-426.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+* Change recommended Java runtime distribution to Eclipse Temurin. ({issue}`18886`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-426.md:9]: This doesn't need a release note item. We don't record changes to documentation in the release notes.

File: docs/src/main/sphinx/release/release-426.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+* Fix event listener incorrectly reporting output columns for `UPDATE`
+  statements with subqueries. ({issue}`18815`)
+* Fix failure when performing an outer join involving geospatial functions in
+  the join clause. ({issue}`18860`)
+* Fix failure when querying partitioned tables with a `WHERE` clause that
+  contains lambda expressions. ({issue}`18865`)
+* Fix index out of bounds exceptions in flat memory data structures when using
```
Comment (martint) [inline docs/src/main/sphinx/release/release-426.md:16]: "flat memory data structures" is an implementation detail. I'd rephrase as:

> Fix failure for `GROUP BY` queries over `map` and `array` types when XXX configuration option is enabled.

File: docs/src/main/sphinx/release/release-426.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
+# Release 426 (5 Sep 2023)
+
+## General
+
+* Add support for `SET SESSION AUTHORIZATION` and `RESET SESSION AUTHORIZATION`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-426.md:5]: Remove the word "syntax". It's unnecessary.

**Review by mosabua — APPROVED** [review]
Looks good now.
